docs/score_releases/index.rst

```diff
@@ -14,6 +14,11 @@
 
 .. _releases:
 
+S-CORE Releases
+===============
+
+A **Release** is a crucial aspects of the project life cycle within the `Eclipse Development Process <https://www.eclipse.org/projects/handbook/#release>`_. It is a officially distributed, versioned, and deployable collection of Project artifacts intended for distribution beyond the Project Developers. It brings together the initial set of core modules, reference integrations, and supporting infrastructure needed to build. For further information about the platform releases and module releases see `release documentation <https://eclipse-score.github.io/process_description//main/process_areas/release_management/release_workproducts.html>`_ in the process description.
```

Comment:
I would leave out the EDP here and then add a paragraph specifically for it. Message should be along the lines of "Eclipse governance and processes apply for S-CORE (as it is an Eclipse project). Therefore S-CORE's release process, in addition to its project-specific steps, must also follow Eclipse requirements, especially a successful, non-expired Progress Review. [link to handbook]"